src/segments/bun.go

```diff
@@ -0,0 +1,34 @@
+package segments
+
+import (
+	"github.com/jandedobbeleer/oh-my-posh/src/platform"
+	"github.com/jandedobbeleer/oh-my-posh/src/properties"
+)
+
+type Bun struct {
+	language
+}
+
+func (b *Bun) Template() string {
+	return languageTemplate
+}
+
+func (b *Bun) Init(props properties.Properties, env platform.Environment) {
+	b.language = language{
+		env:        env,
+		props:      props,
+		extensions: []string{"*.js", "*.ts", "bun.lockb"},
```

Comment:
do we want to use the same extensions as the node segment? Or only trigger on `bun.lock`?

Reply:
I was thinking about this as well. When testing things out, I noticed myself setting `extensions = ["bun.lockb"]` anyway. 
I more or less just copied what we have for `Deno`. Technically you could use the bun cli outside of a bun project.
I don't really have any strong preferences here. Personally I think just having the `bun.lockb` is fine.

Reply:
I would indeed keep `bun.lockb` only as the default, people can override as they see fit.

Reply:
@Joxtacy if you adjust this part, we're ready to merge 👍🏻

Reply:
Done! :) If you'd just like to double check to make sure I didn't miss anything, I'd be happy as well.

Reply:
already did, and if not, that's just a small PR away 😄